### PR TITLE
Fix some sidebar styling issues and clean up sidebar styles

### PR DESCRIPTION
**Pull request by bduffany**
Fixes:
* Fix bottom spacing between profile picker (at bottom) and the bottom edge of the sidebar, which was reproducible in dense mode only
* Fix issue that icons in the group picker become squished when the group name is too long

Refactor:
* Avoid specifying margins where possible and instead use the `gap` property which now has full support in all major browsers: https://caniuse.com/gap. The gap property is cleaner than margins because it allows setting a single CSS property on the parent, rather than having to awkwardly apply margins to all children except the last child.
* Avoid specifying negative margins where possible.
* Avoid non-symmetric padding and margins where possible -- make it so that the top padding almost always matches the bottom padding and left padding almost always matches the right padding.
* Remove `.dense` specific styles where possible in favor of setting a CSS variable that controls the base padding.

Tested in Linux/Chrome, macOS/Chrome, macOS/Safari, across all combinations of {mobile, dense mode, sidebar footer expanded}.

---

**Version bump**: Patch

### CODE REVIEW

File: enterprise/app/sidebar/sidebar.css [hunk 1]
```diff
@@ -172,11 +211,14 @@
   opacity: 1;
 }
 
+.sidebar .org-picker-item svg {
```
Comment (siggisim) [inline enterprise/app/sidebar/sidebar.css:214]: should we apply this to all icons?

**Review by siggisim — APPROVED** [review]

File: enterprise/app/sidebar/sidebar.tsx [hunk 2]
```diff
@@ -280,6 +280,7 @@ export default class SidebarComponent extends React.Component {
                   </div>
                 )}
               </div>
+              <hr />
```
Comment (siggisim) [inline enterprise/app/sidebar/sidebar.tsx:283]: how does this look?

File: enterprise/app/sidebar/sidebar.css (hunk 1 above)
Comment (bduffany) [reply enterprise/app/sidebar/sidebar.css]: sg, done

File: enterprise/app/sidebar/sidebar.tsx (hunk 2 above)
Comment (bduffany) [reply enterprise/app/sidebar/sidebar.tsx:283]: Same as before -- I added an override to remove the default user agent styles:

image

**Pull request merged**